external/kafka/src/test/java/org/apache/spark/streaming/kafka/JavaDirectKafkaStreamSuite.java

```diff
@@ -41,24 +41,31 @@
 
 public class JavaDirectKafkaStreamSuite implements Serializable {
   private transient JavaStreamingContext ssc = null;
-  private transient KafkaStreamSuiteBase suiteBase = null;
+  private transient KafkaTestUtils kafkaTestUtils = null;
 
   @Before
   public void setUp() {
-      suiteBase = new KafkaStreamSuiteBase() { };
-      suiteBase.setupKafka();
-      System.clearProperty("spark.driver.port");
-      SparkConf sparkConf = new SparkConf()
-              .setMaster("local[4]").setAppName(this.getClass().getSimpleName());
-      ssc = new JavaStreamingContext(sparkConf, Durations.milliseconds(200));
+    kafkaTestUtils = new KafkaTestUtils();
+    kafkaTestUtils.setup();
+    System.clearProperty("spark.driver.port");
+    SparkConf sparkConf = new SparkConf()
+      .setMaster("local[4]").setAppName(this.getClass().getSimpleName());
+    ssc = new JavaStreamingContext(sparkConf, Durations.milliseconds(200));
   }
 
   @After
   public void tearDown() {
+    if (ssc != null) {
       ssc.stop();
       ssc = null;
-      System.clearProperty("spark.driver.port");
-      suiteBase.tearDownKafka();
+    }
+
+    System.clearProperty("spark.driver.port");
```

Comment:
No need for clearing other places in this PR. We will clear them up as we
encounter them.

On Thu, Apr 9, 2015 at 1:45 AM, Saisai Shao <notifications@github.com>
wrote:

> In
> external/kafka/src/test/java/org/apache/spark/streaming/kafka/JavaDirectKafkaStreamSuite.java
> <https://github.com/apache/spark/pull/4961#discussion_r28044578>:
>
> >        ssc.stop();
> >        ssc = null;
> > -      System.clearProperty("spark.driver.port");
> > -      suiteBase.tearDownKafka();
> > +    }
> > +
> > +    System.clearProperty("spark.driver.port");
>
> I think this is a quite old code before we have SparkConf, there are some
> places in the unit test still have such code, do I need to clean them all?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/apache/spark/pull/4961/files#r28044578>.
>